File: JavaScript/프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let sum = brown + yellow;
    let width = 0;
    let height = 0;
    for(i = 0 ; i < sum ; i++){
        width = i;
        height = Math.floor(sum / i);
        if (width >= height && (width-2) * (height-2) == yellow && width*height - yellow == brown){
            break;
        }
    }
    return [width,height]
}

File: Java/프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = {0,0};
        int sum = brown + yellow;
        int width,height = 0;
        for(int i = 1 ; i < sum ; i ++){
            width = i;
            height = sum / i ;
            if((width - 2) * ( height - 2 ) == yellow){
                if(width < height){
                    answer[0] = height;
                    answer[1] = width;
                }
                else{
                    answer[0] = width;
                    answer[1] = height;
                }
            }
        }
        return answer;
    }
}